ext/fileinfo/libmagic/softmagic.c

```diff
@@ -512,6 +512,7 @@ check_fmt(struct magic_set *ms, const char *fmt)
 # if defined(__aiws__) || defined(_AIX)
 #  define strndup aix_strndup	/* aix is broken */
```

Comment:
There's no point in defining this just to unset it later. I assume the implementation of `strndup` should've been in the `#else` block.

Reply:
I ve tried on my old centos 7 instance (which genuinely is old enough for not having its glibc with strndup) and the build passes without error tough. So you re right @iluuu1994 here anyway, it should be in a else block but even simpler I think it can be removed, estrndup is used instead anyway it seems what do you think ?

Reply:
Oh, it's not even used. In that case removing it is definitely preferable.

Reply:
I just want to fix the compilation problem and make it pass. You have found the root problem. I tried removing the whole definition. Passed perfectly. Therefore, follow-up to your official handling of this issue.

Reply:
In the case @midoks still want to pursue this PR, just realised this fileinfo extension relies on patch appliance. In that case [in here](https://github.com/php/php-src/blob/master/ext/fileinfo/libmagic.patch).

Reply:
Might even be better to provide the patch [upstream](https://github.com/file/file); we could then update to the new libmagic version.